### PR TITLE
Update: activity feed error message to only show for certain error codes

### PR DESCRIPTION
**Pull request by DanDeMicco**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @DanDeMicco has signed the CLA. Thanks for the pull request!

File: src/api/Feed.js [hunk 1]
```diff
@@ -30,6 +30,9 @@ import { sortFeedItems } from '../util/sorter';
 const DEFAULT_START = 0;
 const DEFAULT_END = 1000;
 const CONFLICT_CODE = 409;
+const UNAUTHORIZED_CODE = 401;
```
Comment (priyajeet) [inline src/api/Feed.js]: Should ideally be in HTTP codes section in constants.js

File: src/api/Feed.js [hunk 2]
```diff
@@ -178,10 +181,7 @@ class Feed extends Base {
             this.commentsAPI.offsetGet(
                 this.id,
                 resolve,
-                err => {
-                    this.errorCallback(err);
-                    resolve();
-                },
+                this.fetchFeedItemErrorCallback.bind(this, resolve),
```
Comment (priyajeet) [inline src/api/Feed.js]: can you change this to the arrow function style? so that we dont need binding

File: src/api/__tests__/Feed-test.js [hunk 3]
```diff
@@ -1146,4 +1146,33 @@ describe('api/Feed', () => {
...
+        });
+
+        test('should call the error callback if error is internal server error', () => {
+            feed.fetchFeedItemErrorCallback(errorCb, {
+                status: 500,
+            });
+            expect(feed.errorCallback).toHaveBeenCalled();
+        });
+
+        test('should call the error callback if error is forbidden or another error', () => {
```
Comment (jstoffan) [inline src/api/__tests__/Feed-test.js]: `should not call`?

File: src/constants.js [hunk 4]
```diff
@@ -250,3 +250,9 @@ export const PLACEHOLDER_USER = { type: 'user', id: '2', name: '' };
 
 /* ------------------ Integrations ------------------------- */
 export const APP_INTEGRATION = 'app_integration';
+
+/* ------------------ Status Codes ------------------------- */
+export const CONFLICT_CODE = 409;
```
Comment (jstoffan) [inline src/constants.js]: There is already a section for this above on line 240.

File: src/api/Feed.js [hunk 5]
```diff
@@ -232,15 +229,38 @@ class Feed extends Base {
...
                 params: requestData,
             });
         });
     }
 
+    /**
+     * Error callback for fetching feed items.
+     * Should only call the error callback if the response is a 401, 429 or >= 500
+     *
+     * @param {Object} e - the axios error
```
Comment (ConradJChan) [inline src/api/Feed.js]: param order?

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #630

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/constants.js (hunk 4 above)
Comment (priyajeet) [reply src/constants.js]: + naming convention should be inverted

File: src/api/Feed.js [hunk 6]
```diff
@@ -232,15 +229,38 @@ class Feed extends Base {
...
+        const { status } = e;
+        if (
+            status === RATE_LIMIT_CODE ||
+            status === UNAUTHORIZED_CODE ||
+            status >= INTERNAL_SERVER_ERROR_CODE
+        ) {
+            this.errorCallback(e);
+        }
+
+        if (typeof cb === 'function') {
```
Comment (priyajeet) [inline src/api/Feed.js]: You can also default it to noop if you want

File: src/api/Feed.js (hunk 2 above)
Comment (jstoffan) [inline src/api/Feed.js]: I think @priyajeet mentioned using arrow functions in a previous comment, but otherwise LGTM.

**Pull request merged**